### PR TITLE
Add example for ML cluster

### PR DESCRIPTION
**Pull request by tpdownes**
Add an example blueprint (and integration test) that provisions an HPC cluster running the Slurm scheduler. Using Packer, each VM within the cluster has the machine learning frameworks [PyTorch] and [TensorFlow] pre-installed. An example PyTorch benchmark job is pre-staged using a data runner. The cluster has 2 partitions showcasing recent Google Cloud GPU families:

- [A2 family VMs][a2] with the NVIDIA A100 GPU accelerator
- [G2 family VMs][g2] with the NVIDIA L4 GPU accelerator

[a2]: https://cloud.google.com/compute/docs/gpus#a100-gpus
[g2]: https://cloud.google.com/compute/docs/gpus#l4-gpus

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -861,6 +860,52 @@ credentials for the created cluster_ and _submit a job calling `nvidia_smi`_.
...
+```shell
+cp /var/tmp/torch_test.* .
+sbatch -N 1 torch_test.sh
+```
+
+When you are done, clean up the resources in reverse order of creation:
+
+```text
+terraform -chdir=image-builder-001/cluster destroy --auto-approve
+terraform -chdir=image-builder-001/primary destroy --auto-approve
```
Comment (cboneti) [inline examples/README.md:904]: how do they destroy the image? (gcloud?)

File: tools/cloud-build/daily-tests/ansible_playbooks/multigroup-integration-test.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  - name: Deploy from deployment directory
+    block:
+    - name: Execute ghpc deploy
+      register: deployment
+      changed_when: deployment.changed
+      ansible.builtin.command: ./ghpc deploy {{ deployment_name }} --auto-approve
+      args:
+        chdir: "{{ workspace }}"
+      environment:
+        TF_IN_AUTOMATION: "TRUE"
```
Comment (cboneti) [inline tools/cloud-build/daily-tests/ansible_playbooks/multigroup-integration-test.yml:33]: It would have been great if you could make this call the slurm integration tests from here, as it could test that the cluster actually worked.

**Review by cboneti — COMMENTED** [review]
Please see my comments. 
Also, optional: consider moving this blueprint to core and removing the experimental badge.

**Review by cboneti — COMMENTED** [review]
Please see my comments. 
Also, optional: consider moving this blueprint to core and removing the experimental badge.

**Review by cboneti — COMMENTED** [review]
Please see my comments. 
Also, optional: consider moving this blueprint to core and removing the experimental badge.

File: examples/README.md (hunk 1 above)
Comment (tpdownes) [reply examples/README.md:904]: Flexible instructions for deleting the image (which is uniquely named) via the console are provided in the lines directly below this section of text.

File: tools/cloud-build/daily-tests/ansible_playbooks/multigroup-integration-test.yml (hunk 2 above)
Comment (tpdownes) [reply tools/cloud-build/daily-tests/ansible_playbooks/multigroup-integration-test.yml:33]: I want to add that functionality in the next update to these tests, which intention of further reducing code duplication. The approach of getting the login node IP (gcloud command piped to a regexp search) is fragile and broke recently and I don't want to expand its radius until we resolve it. I realzie that the task I captured for that work is related but doesn't capture it fully so I will update the task.

(The more solid solution is to get the IP/DNS name from `terraform output`).

File: examples/README.md (hunk 1 above)
Comment (cboneti) [reply examples/README.md:904]: I can't seem to find it... what am I missing?

Comment (tpdownes) [reply examples/README.md:904]: Oh, you are right. It appears in the (similar) image-builder example but not here. Will fix.

Comment (tpdownes) [reply examples/README.md]: See a6d72f9cf048bf36a4e8a0ff49dd085492754754

File: examples/README.md [hunk 3]
```diff
@@ -861,6 +860,56 @@ credentials for the created cluster_ and _submit a job calling `nvidia_smi`_.
...
+
+When you are done, clean up the resources in reverse order of creation:
+
+```text
+terraform -chdir=ml-example/cluster destroy
+terraform -chdir=ml-example/primary destroy
+```
+
+Finally, browse to the [Cloud Console][console-images] to delete your custom
+image. It will be named beginning with `ml-slurm` followed by a date and
```
Comment (cboneti) [inline examples/README.md:908]: I hate to nit pick here: ml-slurm can change if the user changes their deployment name. Should we not tie this to the deployment name?

**Review by cboneti — APPROVED** [review]
consider changing the readme to tie to the deployment name rather than `ml-slurm`, but also, consider this approved.
